pkg/scalers/prometheus_scaler_test.go

```diff
@@ -376,37 +375,43 @@ func TestPrometheusScalerCustomHeaders(t *testing.T) {
 }
 
 func TestPrometheusScalerExecutePromQueryParameters(t *testing.T) {
-	for _, testData := range testPromQueryResult {
-		t.Run(testData.name, func(t *testing.T) {
-			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, _ *http.Request) {
-				writer.WriteHeader(testData.responseStatus)
-
-				if _, err := writer.Write([]byte(testData.bodyStr)); err != nil {
-					t.Fatal(err)
-				}
-			}))
+	testData := prometheusQromQueryResultTestData{
+		name:             "no values",
+		bodyStr:          `{"data":{"result":[]}}`,
+		responseStatus:   http.StatusOK,
+		expectedValue:    0,
+		isError:          false,
+		ignoreNullValues: true,
+	}
+	queryParametersValue := map[string]string{
+		"first":  "foo",
+		"second": "bar",
+	}
+	server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		for queryParameterName, queryParameterValue := range queryParametersValue {
+			queryParameter := request.URL.Query()
+			queryParameter.Add(queryParameterName, queryParameterValue)
+		}
 
-			scaler := prometheusScaler{
-				metadata: &prometheusMetadata{
-					serverAddress:    server.URL,
-					ignoreNullValues: testData.ignoreNullValues,
-					unsafeSsl:        testData.unsafeSsl,
-				},
-				httpClient: http.DefaultClient,
-				logger:     logr.Discard(),
-			}
+		
+		writer.WriteHeader(testData.responseStatus)
+		if _, err := writer.Write([]byte(testData.bodyStr)); err != nil {
+			t.Fatal(err)
+		}
```

Comment:
You aren't testing the received query parameters, This is just adding more but you aren't testing them.
You have an example in RabbitMQ tests about how it can be done: https://github.com/kedacore/keda/blob/main/pkg/scalers/rabbitmq_scaler_test.go#L704-L707
It's a bit "rustic" way, but it works. You can also check the query values one by one getting them and validating that they are within the `queryParametersValue` slice

Reply:
Test implemented as sugested